### PR TITLE
Add Quickbooks2016

### PR DESCRIPTION
**Pull request by sscotth**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask-versions/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/quickbooks2016.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'quickbooks2016' do
+  version '2016'
+  sha256 :no_check # required as upstream package is updated in-place
```
Comment (commitay) [inline Casks/quickbooks2016.rb]: Please add the `sha256`

**Comment by sscotth** [comment]
I'd like to add more detail to the `:version` to make changes more explicit.

Would this work:
```rb
version '2016,17.2.27 R28,682'
# ...
app "QuickBooks #{version.before_comma}.app"
`

or should we hardcode the year, and just use the version properly:

```rb
version '17.2.27 R28,682'
# ...
app "QuickBooks 2016.app"
`

![screen shot 2018-09-27 at 11 57 11 am](https://user-images.githubusercontent.com/1010488/46162265-5b9baf00-c24d-11e8-9c9a-328d166fddbf.png)

![screen shot 2018-09-27 at 11 57 13 am](https://user-images.githubusercontent.com/1010488/46162269-5d657280-c24d-11e8-99f6-75ad7e4cfb67.png)

**Comment by reitermarkus** [comment]
> I'd like to add more detail to the `version` to make changes more explicit.

Since the download URL doesn't contain anything more than `2016`, that wouldn't bring any benefit.

**Comment by sscotth** [comment]
Except when updates happen, the url doesn't change, only the sha. So if there is no change to `version` you would have to [confirm the new checksum to help rule out malicious tampering](https://github.com/homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256).

**Comment by reitermarkus** [comment]
If you can find an appcast, we can use the full version string.

**Comment by reitermarkus** [comment]
If you can find an appcast, we can use the full version string.

**Comment by vitorgalvao** [comment]
Do we still want this?

**Pull request merged**